File: app/src/main/java/com/stevenfrew/beatprompter/midi/SongDisplayInTask.java
package com.stevenfrew.beatprompter.midi;

import android.util.Log;

import com.stevenfrew.beatprompter.Task;

public class SongDisplayInTask extends Task
{
    SongDisplayInTask()
    {
        super(false);
    }
    public void doWork()
    {
        try {
            while ((MIDIController.mMIDISongDisplayInQueue.take() != null) && (!getShouldStop())) {
                Log.d(MIDIController.MIDI_TAG,"Discarding message intended for song display mode");
                // Do nothing. These messages aren't meant for this activity.
            }
        } catch (InterruptedException ie) {
            Log.d(MIDIController.MIDI_TAG, "Interrupted while attempting to retrieve MIDI in message.", ie);
        }
    }
}

File: app/src/main/kotlin/com/stevenfrew/beatprompter/midi/SongDisplayInTask.kt
package com.stevenfrew.beatprompter.midi

import android.util.Log
import com.stevenfrew.beatprompter.Task

class SongDisplayInTask internal constructor() : Task(false) {
    override fun doWork() {
        try {
            while (MIDIController.mMIDISongDisplayInQueue.take() != null && !shouldStop) {
                Log.d(MIDIController.MIDI_TAG, "Discarding message intended for song display mode")
                // Do nothing. These messages aren't meant for this activity.
            }
        } catch (ie: InterruptedException) {
            Log.d(MIDIController.MIDI_TAG, "Interrupted while attempting to retrieve MIDI in message.", ie)
        }
    }
}